feat: [CDS-70965]: Update sample-policy-use-case.md (#2282)
harness/developer-hub: docs/platform/Governance/14_Policy-as-code/sample-policy-use-case.md

```

> #### Enforce execution from default branch only if user is not part of a specific userGroup for a remote Pipeline 

Reviewer: ```suggestion
#### Enforce remote pipeline execution from the default branch only if the user is not part of a specific user group
```

> Users part of a specific user group can run pipeline from any branch, but other users can run only from default branch for remote pipelines

Reviewer: ```suggestion
Users in a specific user group can run remote pipelines from any branch, but others can only run pipelines from the default branch.
```
